assertsprocessor/metrics.go

```diff
@@ -0,0 +1,166 @@
+package assertsprocessor
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/collectors"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.uber.org/zap"
+	"net/http"
+	"regexp"
+	"strings"
+	"time"
+)
+
+type metricHelper struct {
+	logger                *zap.Logger
+	config                *Config
+	prometheusRegistry    *prometheus.Registry
+	latencyHistogram      *prometheus.HistogramVec
+	attributeValueRegExps *map[string]*regexp.Regexp
+}
+
+func (p *metricHelper) compileSpanFilterRegexps() error {
+	p.logger.Info("consumer.Start compiling regexps")
+	for attName, matchExpString := range *p.config.AttributeExps {
+		compile, err := regexp.Compile(matchExpString)
+		if err != nil {
+			return err
+		}
+		(*p.attributeValueRegExps)[attName] = compile
+	}
+	p.logger.Debug("consumer.Start compiled regexps successfully")
+	return nil
+}
+
+// Returns true if a span matches the span selection criteria
+func (p *metricHelper) shouldCaptureMetrics(span ptrace.Span) bool {
+	if len(*p.attributeValueRegExps) > 0 {
+		spanAttributes := span.Attributes()
+		for attName, matchExp := range *p.attributeValueRegExps {
+			value, found := spanAttributes.Get(attName)
+			if !found {
+				return false
+			}
+			//p.logger.Info("Found Span Attribute",
+			//	zap.String(attName, value.AsString()))
+
+			valueMatches := matchExp.String() == value.AsString() || matchExp.MatchString(value.AsString())
+			//p.logger.Info("Value Regexp Result",
+			//	zap.String("regexp", matchExp.String()),
+			//	zap.Bool("result", valueMatches))
+			if !valueMatches {
+				return false
+			}
+		}
+		return true
```

Comment:
Why return `true` when there is no match?

Reply:
> Why return `true` when there is no match?
`true` will be returned only when all the expected attributes are present and match. Even if one attribute doesn't match, we return false. If no span attribute filter regexps are configured, we return `false`. It's all-or-none right now. I now think that it should probably be `atleast-one`.

Reply:
all-or-none seems more intuitively. Normally when we have a list of conditions, we think it's an `AND`.